text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,186 @@
+This proposal introduces a new value with the key `th` as an alternative to the `p` value in the previous specification.
+The `p` value is limited to powers of two, while the `th` value in this proposal supports a large range of values.
+This proposal allows for the continued expression of randomness using `r-value` as specified there using the key `r`.
+To distinguish the cases, this proposal uses the key `rv`.
+
+In the general case, in order to make consistent sampling decisions across the entire path of the trace, two values MUST be present in the `SpanContext`:
+
+1. A _random_ (or pseudo-random) 56-bit value, called `R` below.
+2. A 56-bit _rejection threshold_ (or just "threshold") as expressed in the TraceState, called `T` below. `T` represents the maximum threshold that was applied in all previous consistent sampling stages. If the current sampling stage applies a greater-valued threshold than any stage before, it MUST update (increase) the threshold correspondingly.
+
+Here is an example involving three participants `A`, `B`, and `C`:
+
+`A` -> `B` -> `C`
+
+where -> indicates a parent -> child relationship.
+
+`A` uses consistent probability sampling with a sampling probability of 0.25 (this corresponds to a rejection probability of .75).
+`B` uses consistent probability sampling with a sampling probability of 0.5.
+`C` uses a parent-based sampler.
+
+When `A` samples a span, its outgoing traceparent will have the 'sampled' flag SET and the 'th' in its outgoing tracestate will be set to `0xc0_0000_0000_0000`.
+When `A` does not sample a span, its outgoing traceparent will have the 'sampled' flag UNSET but the 'th' in its outgoing tracestate will still be set to 0xc0_0000_0000_0000.
+When B samples a span, its outgoing traceparent will have the 'sampled' flag SET and the 'th' in its outgoing tracestate will be set to 0x80_0000_0000_0000.
+C (being a parent based sampler) samples a span purely based on its parent (B in this case), it will use the sampled flag to make the decision. Its outgoing 'th' value will continue to reflect what it got from B (0x80_0000_0000_0000), and this is useful to understand its adjusted count.
+
+This design requires that as a given span progresses along its collection path, `th` is non-decreasing (and, in particular, must be increased at stages that apply lower sampling probabilities).
+It does not, however, restrict a span's initial `th` in any way (e.g., relating it to that of its parent, if it has one).
+It is acceptable for B to have a lesser initial `th` than A has. It would not be ok if some later-stage sampler decreased A's `th`.
+
+The system has the following invariant:
+
+`(R >= T) = sampled flag)`
+
+The sampling decision is propagated with the following algorithm:
+
+* If the `th` key is not specified, Always Sample.
+* Else derive `T` by parsing the `th` key as a hex value as described below.
+* If `T` is 0, Always Sample. This implies that non-probabilistic sampling is taking place.
```

Comment:
I believe our intention was that missing `th` key implies non-probabilistic sampling. `T` equal to zero means 100% (probabilistic) sampling.